.github/workflows/build.yaml

```
@@ -0,0 +1,34 @@
+name: Build
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+          cache: gradle
+
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Build the project
+        run: ./gradlew build --stacktrace
```

Minor thing:
```suggestion
        run: ./gradlew build --stacktrace --warning-mode=fail
```

Nice to keep ontop of Gradle deprecations.